Mathlib/Dynamics/TopologicalEntropy/DynamicalUniformity.lean

```diff
@@ -0,0 +1,131 @@
+/-
+Copyright (c) 2024 Damien Thomine. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damien Thomine, Pietro Monticone
+-/
+import Mathlib.Order.Interval.Finset.Nat
+import Mathlib.Order.OmegaCompletePartialOrder
+import Mathlib.Topology.UniformSpace.Basic
+
+/-!
+# Dynamical uniformities
+Bowen-Dinaburg's definition of topological entropy of a transformation `T` in a metric space
+`(X, d)` relies on the so-called dynamical balls. These balls are sets
+`B (x, ε, n) = { y | ∀ k < n, d(T^[k] x, T^[k] y) < ε }`.
+
+We implement Bowen-Dinaburg's definitions in the more general context of uniform spaces. Dynamical
+balls are replaced by what we call dynamical uniformities. This file collects all general lemmas
+about these objects.
+
+## Main definitions
+- `DynamicalUni`: dynamical uniformity associated with a given transformation `T`, uniformity `U`
+and time `n`.
+
+## Tags
+entropy
+
+## TODO
+Once #PR14718 has passed, add product of uniformities.
+
+In the context of (pseudo-e)metric spaces, relate the usual definition of dynamical balls with
+these dynamical uniformities.
+-/
+
+namespace DynamicalUniformity
+
+open Prod Set Uniformity UniformSpace
+
+variable {X : Type*}
+
+/-- A dynamical uniform neighborhood is the uniform space version of dynamical balls.-/
+def DynamicalUni (T : X → X) (U : Set (X × X)) (n : ℕ) : Set (X × X) :=
+  ⋂ k < n, (map T T)^[k] ⁻¹' U
+
+theorem dynamical_uni_inter_Ico (T : X → X) (U : Set (X × X)) (n : ℕ) :
```

Comment:
```suggestion
theorem dynEntourage_eq_inter_Ico (T : X → X) (U : Set (X × X)) (n : ℕ) :
```
Also, all the theorems should be lemmas: `theorem` is only for important results, with docstrings.

Reply:
Done.